pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -134,39 +154,36 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 }
 
 func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) error {
-	// fetch the equivelant pipeline for this pipelinerun Run
 	p, err := c.pipelineLister.Pipelines(pr.Namespace).Get(pr.Spec.PipelineRef.Name)
 	if err != nil {
 		c.Logger.Errorf("%q failed to Get Pipeline: %q",
 			fmt.Sprintf("%s/%s", pr.Namespace, pr.Name),
 			fmt.Sprintf("%s/%s", pr.Namespace, pr.Spec.PipelineRef.Name))
 		return nil
 	}
-	pipelineTasks, err := p.GetTasks(func(namespace, name string) (*v1alpha1.Task, error) {
-		return c.taskLister.Tasks(namespace).Get(name)
-	})
-	if err != nil {
-		return fmt.Errorf("error getting Tasks for Pipeline %s, Pipeline is invalid!: %s", p.Name, err)
-	}
-	pipelineTaskName, trName, err := resources.GetNextPipelineRunTaskRun(
+	state, err := resources.GetPipelineState(
+		func(namespace, name string) (*v1alpha1.Task, error) {
+			return c.taskLister.Tasks(namespace).Get(name)
+		},
 		func(namespace, name string) (*v1alpha1.TaskRun, error) {
 			return c.taskRunLister.TaskRuns(namespace).Get(name)
 		},
-		p, pr.Name)
+		p, pr.Name,
+	)
 	if err != nil {
-		return fmt.Errorf("error getting next TaskRun to create for PipelineRun %s: %s", pr.Name, err)
+		return fmt.Errorf("error getting Tasks for Pipeline %s, Pipeline may be invalid!: %s", p.Name, err)
```

Comment:
s/error getting Tasks for Pipeline/error getting Tasks or TaskRuns for Pipeline/

Reply:
good call!